deployability/modules/allocation/aws/provider.py

```diff
@@ -103,7 +107,10 @@ def _create_instance(cls, base_dir: Path, params: CreationPayload, config: AWSCo
         else:
             credentials.key_path = (os.path.splitext(ssh_key)[0])
 
-        return AWSInstance(instance_dir, instance_id, credentials, config.user)
+        if config.host_id:
+            return AWSInstance(instance_dir, instance_id, credentials, host_id, config.user)
+        else:
+            return AWSInstance(instance_dir, instance_id, credentials, None, config.user)
```

Comment:
In this case the if sentence is probably not required as the value sent to `AWSInstance` when there is no `config.host_id` will be in `None` anyways because the `host_id` default value is `None`
```
host_id: str | None = None
```

```suggestion
        return AWSInstance(instance_dir, instance_id, credentials, host_id, config.user)
```